src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt

```diff
@@ -547,6 +566,21 @@ open class IndexReplicationTask(id: Long, type: String, action: String, descript
                 } else {
                     metadataUpdate = null
                 }
+                val options = IndicesOptions.strictSingleIndexNoExpandForbidClosed()
+                var gmr = GetMappingsRequest().indices(this.leaderIndex.name).indicesOptions(options)
+                var mappingResponse = remoteClient.suspending(remoteClient.admin().indices()::getMappings, injectSecurityContext = true)(gmr)
+                var leaderMappingSource = mappingResponse.mappings.get(this.leaderIndex.name).source().toString()
+                val leaderProperties = mappingResponse.mappings().get(this.leaderIndex.name).sourceAsMap().toMap().get("properties") as Map<String,Any>
+                gmr = GetMappingsRequest().indices(this.followerIndexName).indicesOptions(options)
+                mappingResponse = client.suspending(client.admin().indices()::getMappings, injectSecurityContext = true)(gmr)
+                val followerProperties = mappingResponse.mappings().get(this.followerIndexName).sourceAsMap().toMap().get("properties") as Map<String,Any>
+                for(iter in followerProperties) {
+                    if(leaderProperties.containsKey(iter.key) && leaderProperties.getValue(iter.key).toString()!=(iter.value).toString()){
```

Comment:
Can we check, if there is a better way to identify these fields?

Reply:
if there are multiple mismatches here, we'll end up making multiple requests to update the mapping. Can you instead, just maintain a flag like `mapping_update_required` and update after the loop ends.